float32 e_s1
float32 e_s2
float32 e_s3
